.gitignore

```diff
@@ -1,22 +1,43 @@
-/bin
-/build
-/dist
-/freenet-*
-/javadoc
-/junitvmwatcher*
-/lib
-/generator/js/lib
-/generator/js/tests/lib
-/generator/js/tests/bin
-/generator/js/gwt-unitCache/
-/run
-.classpath
-*.swp
+freenet-*
+junitvmwatcher*
+lib/
+contrib/
+generator/js/lib
+generator/js/tests/lib
+generator/js/tests/bin
+generator/js/gwt-unitCache/
+
+# Intellij
+.idea/
 *.iml
 *.ipr
 *.iws
-.*
-*~
-\#*\#
-/override.properties
-/gradle.properties
+*.swp
```

Comment:
While I totally agree that the repository should not contain any files that are not relevant, I am of the opinion that we only need to exclude files in `.gitignore` that are actually created by the build system. It’s every developer’s own responsibility to not check in their editor’s, IDE’s, or OS’s temp or backup files that might end up in the working copy of the repository. (Git can use a global `.gitignore` file, too.)